File: frontend/src/api/git-integration.ts
import api from './index'

export interface GitIntegration {
  id: number
  project_id: number
  provider: string
  repo_url: string
  repo_name: string
  access_token?: string
  webhook_secret?: string
  active: boolean
  sync_prs: boolean
  sync_commits: boolean
  sync_branches: boolean
  created_at: string
  updated_at: string
}

export interface GitIssueLink {
  id: number
  issue_id: number
  git_type: string
  git_id: string
  git_url: string
  git_title: string
  git_state: string
  git_author: string
  git_branch: string
  created_at: string
  updated_at: string
}

export async function createGitIntegration(
  workspaceId: number,
  projectId: number,
  data: {
    provider: string
    repo_url: string
    repo_name: string
    access_token?: string
    webhook_secret?: string
  }
): Promise<GitIntegration> {
  const response = await api.post(`/workspaces/${workspaceId}/git-integration?project_id=${projectId}`, data)
  return response.data
}

export async function getGitIntegration(
  workspaceId: number,
  projectId: number
): Promise<GitIntegration> {
  const response = await api.get(`/workspaces/${workspaceId}/git-integration?project_id=${projectId}`)
  return response.data
}

export async function updateGitIntegration(
  workspaceId: number,
  projectId: number,
  data: Partial<GitIntegration>
): Promise<GitIntegration> {
  const response = await api.put(`/workspaces/${workspaceId}/git-integration?project_id=${projectId}`, data)
  return response.data
}

export async function deleteGitIntegration(
  workspaceId: number,
  projectId: number
): Promise<void> {
  await api.delete(`/workspaces/${workspaceId}/git-integration?project_id=${projectId}`)
}

export async function getIssueGitLinks(
  workspaceId: number,
  issueId: number
): Promise<GitIssueLink[]> {
  const response = await api.get(`/workspaces/${workspaceId}/issues/${issueId}/git-links`)
  return response.data
}

File: backend/internal/model/git_integration.go
package model

import "time"

type GitIntegration struct {
	ID            uint64     `gorm:"primaryKey;autoIncrement" json:"id"`
	ProjectID     uint64     `gorm:"index;not null" json:"project_id"`
	Provider      string     `gorm:"type:varchar(20);not null" json:"provider"`
	RepoURL       string     `gorm:"type:text" json:"repo_url"`
	RepoName      string     `gorm:"type:varchar(255)" json:"repo_name"`
	AccessToken   string     `gorm:"type:text" json:"access_token"`
	WebhookSecret string     `gorm:"type:varchar(255)" json:"webhook_secret"`
	Active        bool       `gorm:"default:true" json:"active"`
	SyncPRs       bool       `gorm:"default:true" json:"sync_prs"`
	SyncCommits   bool       `gorm:"default:true" json:"sync_commits"`
	SyncBranches  bool       `gorm:"default:false" json:"sync_branches"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

func (GitIntegration) TableName() string {
	return "git_integrations"
}

type GitIssueLink struct {
	ID           uint64     `gorm:"primaryKey;autoIncrement" json:"id"`
	IssueID      uint64     `gorm:"index;not null" json:"issue_id"`
	GitType      string     `gorm:"type:varchar(20);not null" json:"git_type"`
	GitID        string     `gorm:"type:varchar(255);not null" json:"git_id"`
	GitURL       string     `gorm:"type:text" json:"git_url"`
	GitTitle     string     `gorm:"type:text" json:"git_title"`
	GitState     string     `gorm:"type:varchar(20)" json:"git_state"`
	GitAuthor    string     `gorm:"type:varchar(255)" json:"git_author"`
	GitBranch    string     `gorm:"type:varchar(255)" json:"git_branch"`
	IntegrationID uint64    `gorm:"index" json:"integration_id"`
	CreatedAt    time.Time  `json:"created_at"`
	UpdatedAt    time.Time  `json:"updated_at"`
}

func (GitIssueLink) TableName() string {
	return "git_issue_links"
}